ext/dom/xpath.c

```diff
@@ -446,6 +446,51 @@ PHP_METHOD(DOMXPath, registerPhpFunctionNS)
 	);
 }
 
+/* {{{ */
+PHP_METHOD(DOMXPath, quote) {
+	const char *input;
+	size_t input_len;
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s", &input, &input_len) == FAILURE) {
+		RETURN_THROWS();
+	}
+	if (memchr(input, '\'', input_len) == NULL) {
+		zend_string *const output = zend_string_alloc(input_len + 2, 0);
```

Comment:
and what should it do if it actually overflows and the allocation fail? RETURN_THROWS?

Reply:
`zend_string_safe_alloc` will do the error handling upon overflow or allocation failure for you. That means that it will never return NULL. You don't have to do anything special.

Reply:
replaced it with safe_alloc.